lib/HeroBonus.h

```diff
@@ -331,6 +330,21 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(SPECIAL_FIXED_VALUE_ENCHANT) /*specialty spell like Melody has, constant spell effect (i.e. 3 luck), additionalInfo = value to fix.*/\
 	BONUS_NAME(TOWN_MAGIC_WELL) /*one-time pseudo-bonus to implement Magic Well in the town*/\
 	BONUS_NAME(LIMITED_SHOOTING_RANGE) /*limits range of shooting creatures, doesn't adjust any other mechanics (half vs full damage etc). val - range in hexes, additional info - optional new range for broken arrow mechanic */\
+	BONUS_NAME(LEARN_BATTLE_SPELL_CHANCE) /*skill-agnostic eagle eye chance. subtype = 0 - from enemy, 1 - TODO: from entire battlefield*/\
+	BONUS_NAME(LEARN_BATTLE_SPELL_LIMIT) /*skill-agnostic eagle eye limit, subtype - school (-1 for all), others TODO*/\
+	BONUS_NAME(PERCENTAGE_DAMAGE_BOOST) /*skill-agnostic archery and offence, subtype is 0 for offence and 1 for archery*/\
+	BONUS_NAME(LEARN_MEETING_SPELL_LIMIT) /*skill-agnostic scholar, subtype is -1 for all, TODO for others (> 0)*/\
+	BONUS_NAME(ROUGH_TERRAIN_DISCOUNT) /*skill-agnostic pathfinding*/\
+	BONUS_NAME(CHARISMA_BONUS) /*skill-agnostic diplomacy*/\
+	BONUS_NAME(BEFORE_BATTLE_REPOSITION) /*skill-agnostic tactics, bonus for allowing tactics*/\
+	BONUS_NAME(BEFORE_BATTLE_REPOSITION_BLOCK) /*skill-agnostic tactics, bonus for blocking opposite tactics. For now donble side tactics is TODO.*/\
+	BONUS_NAME(HERO_EXP_PERCENT) /*skill-agnostic learning, and we can use it as a global effect also*/\
+	BONUS_NAME(UNDEAD_RAISE_CHANCE) /*Chance to raise creatures after battle as undead*/\
+	BONUS_NAME(HERO_MANA_PER_TEN_KNOWLEDGE) /*Percentage rate of translating 10 hero knowledge to mana, used to intelligence and global bonus*/\

```

Comment:
Really weird form for bonus name.
I mean, I get that you need this for learning, but perhaps learning should apply percentage to base? Then you can use integer "25" for base level

Reply:
Sorry, not Learning but Intelligence.

What I am proposing is to give 10 mana/knowledge as global effect as you've done right now (as BASE_NUMBER) but give bonus from Intelligence skill as PERCENT_TO_ALL or PERCENT_TO_BASE type.

This should allow defining this bonus as simply HERO_MANA_PER_KNOWLEDGE

Reply:
@IvanSavenko then artifacts which can give HERO_MANA_PER_KNOWLEDGE as BASE_NUMBER will be affected by intellect, which is not good.

Reply:
1) Why?
2) These mods don't exist right now so it is up to mod maker to decide whether it is good.
3) We can use PERCENT_TO_BASE, allowing mods to use either BASE_NUMBER or ADDITIVE_VALUE, depending on whether they want skill to affect their artifact

Reply:
Maybe I do not get it... but why 10 knowledge? Sounds weird. Its 10 mana for 1 knowledge.

Reply:
KNOWLEDGE_TO_MANA_RATIO?